lib/active_merchant/billing/gateways/elavon.rb

```diff
@@ -152,216 +184,196 @@ def supports_scrubbing?
       end
 
       private
 
-      def add_invoice(form, options)
-        form[:invoice_number] = truncate((options[:order_id] || options[:invoice]), 10)
-        form[:description] = truncate(options[:description], 255)
+      def add_invoice(xml, options)
+        xml.ssl_invoice_number    truncate((options[:order_id] || options[:invoice]), 25)
+        xml.ssl_description       truncate(options[:description], 255)
       end
 
-      def add_approval_code(form, authorization)
-        form[:approval_code] = authorization.split(';').first
+      def add_approval_code(xml, authorization)
+        xml.ssl_approval_code authorization.split(';').first
       end
 
-      def add_txn_id(form, authorization)
-        form[:txn_id] = authorization.split(';').last
+      def add_txn_id(xml, authorization)
+        xml.ssl_txn_id authorization.split(';').last
       end
 
-      def authorization_from(response)
-        [response['approval_code'], response['txn_id']].join(';')
-      end
+      def add_creditcard(xml, creditcard)
+        xml.ssl_card_number   creditcard.number
+        xml.ssl_exp_date      expdate(creditcard)
 
-      def add_creditcard(form, creditcard)
-        form[:card_number] = creditcard.number
-        form[:exp_date] = expdate(creditcard)
+        add_verification_value(xml, creditcard) if creditcard.verification_value?
 
-        add_verification_value(form, creditcard) if creditcard.verification_value?
-
-        form[:first_name] = truncate(creditcard.first_name, 20)
-        form[:last_name] = truncate(creditcard.last_name, 30)
+        xml.ssl_first_name    truncate(creditcard.first_name, 20)
+        xml.ssl_last_name     truncate(creditcard.last_name, 30)
       end
 
-      def add_currency(form, money, options)
+      def add_currency(xml, money, options)
         currency = options[:currency] || currency(money)
-        form[:transaction_currency] = currency if currency && (@options[:multi_currency] || options[:multi_currency])
+        return unless currency && (@options[:multi_currency] || options[:multi_currency])
+
+        xml.ssl_transaction_currency currency
       end
 
-      def add_token(form, token)
-        form[:token] = token
+      def add_token(xml, token)
+        xml.ssl_token token
       end
 
-      def add_verification_value(form, creditcard)
-        form[:cvv2cvc2] = creditcard.verification_value
-        form[:cvv2cvc2_indicator] = '1'
+      def add_verification_value(xml, creditcard)
+        xml.ssl_cvv2cvc2            creditcard.verification_value
+        xml.ssl_cvv2cvc2_indicator  1
       end
 
-      def add_customer_data(form, options)
-        form[:email] = truncate(options[:email], 100) unless empty?(options[:email])
-        form[:customer_code] = truncate(options[:customer], 10) unless empty?(options[:customer])
-        form[:customer_number] = options[:customer_number] unless empty?(options[:customer_number])
```

Comment:
Are these two fields no longer supported? [I see them on this list](https://developer.elavon.com/#/api/eb6e9106-0172-4305-bc5a-b3ebe832f823.rcosoomi/documents?converge-integration-guide/book/api_reference/../../book/api_reference/supported_transaction_input_fields.html), but I'm not entirely sure this doc is for the XML API.

Reply:
Oo good catch. I rashly assumed that there would be tests for these things which would then fail. >.>

Reply:
I hope you've learned your lesson... 😬 

Staring at the code some more, [I did find the `customer_number` inside a different method](https://github.com/activemerchant/active_merchant/pull/3784/files#diff-4ecda340986024e00f9189e071f58976R282).